docs/contributing/development-guide.md

```diff
@@ -368,10 +342,138 @@ all of the GitHub action checks have run.
 
 ```{note}
 The actual code coverage report is
-uploaded on the GitHub action run on `ubuntu` and `Python 3.9`. When that step in the
+uploaded on the GitHub action run on `ubuntu` and `Python 3.11`. When that step in the
 actions completes, the report will be processed and returned to the pull request.
 ```
 
+
+## Tests & the stravalib mock fixture
+
+To run integration tests that ensure stravalib is interacting with API data correctly, Stravalib uses a mock object access through a
+`pytest` fixture `stravalib.tests.integration.strava_api_stub.StravaAPIMock`
+that is based on `responses.RequestsMock`.
+
+This fixture adds a pass-through mock that prevents requests from being made to the Strava API.
+Instead, it creates responses using the endpoint provided and the `swagger.json` file that is found both online and within the `stravalib/src/stravalib/tests/resources/` directory that are based on examples from the published Strava API
+documentation.
+
+:::{tip}
+Example usages of this fixture can be found in the
+{py:mod}`stravalib.tests.integration.test_client` module.
+:::
+
+:::{mermaid}
+
+flowchart TD
+    B["conftest.py"] -- Defines fixture --> C["mock_strava_api fixture"]
+    C -- Calls --> D["StravaAPIMock from strava_api_stub.py module"]
+    D -- Inherits from --> E["requests_mock"]
+    D -- Adds API Call passthru and returns data using: --> G["swagger.json <br>(local or online)"]
+    D -- Calls --> H["_api_method_adapter"]
```

Comment:
I don't think (new) contributors need to know that level of detail. Maybe we can remove this single line?

Reply:
perfect. i simplified the diagram. i'll merge once CI is happy! thank you for the feedback!